### PR TITLE
HPCC-22981 Add environment theme admin preference

### PR DESCRIPTION
**Pull request by miguelvazq**
Fixes: HPCC-22981

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/eclwatch/templates/HPCCPlatformWidget.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
     <div id="${id}BorderContainer" class="hpccMainpage" data-dojo-props="gutters:false, liveSplitters:false" data-dojo-type="dijit.layout.BorderContainer">
         <div id="${id}Titlebar" class="hpccTitlebar" data-dojo-props="region: 'top'" data-dojo-type="dijit.layout.ContentPane">
             <div id="${id}StackController" class="left glow" data-dojo-props="containerId:'${id}TabContainer'" data-dojo-type="dijit.layout.StackController"></div>
-            <div class="searchUserMoreComponents">
+            <div id="searchUserMoreComponents" class="searchUserMoreComponents">
```
Comment (GordonSmith) [inline esp/src/eclwatch/templates/HPCCPlatformWidget.html]: id should have a "${id}" prefix

File: esp/src/src/UserPreferences/EnvironmentTheme.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import * as Utility from "../Utility";
+
+import { globalKeyValStore } from "../KeyValStore";
+
+import * as query from "dojo/query";
+import * as domConstruct from "dojo/dom-construct";
+import * as domStyle from "dojo/dom-style";
+import * as dom from "dojo/dom";
+
+let ws_store = globalKeyValStore();
```
Comment (GordonSmith) [inline esp/src/src/UserPreferences/EnvironmentTheme.ts]: Should be const?

File: esp/src/src/UserPreferences/EnvironmentTheme.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+declare const dojoConfig;
+
+export function checkCurrentState(id, context) {
+    ws_store.get("HPCCPlatformWidget_Toolbar_Color").then(function(val) {
+        if (val) {
+            domStyle.set(id + "Titlebar", {
+                "backgroundColor": val
+            });
+            context.toolbarColor.set("value", val);
```
Comment (GordonSmith) [inline esp/src/src/UserPreferences/EnvironmentTheme.ts]: Is `context` valid here?

File: esp/src/src/UserPreferences/EnvironmentTheme.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            context.textColor = Utility.textColor(val);
+        } else {
+            context.toolbarColor.set("value", "#2196F3");
+            context.textColor = Utility.textColor("#2196F3");
+        }
+    });
+
+    ws_store.get("HPCCPlatformWidget_Toolbar_Text").then(function(val) {
+        let searchUserMoreComponents = dom.byId("searchUserMoreComponents");
```
Comment (GordonSmith) [inline esp/src/src/UserPreferences/EnvironmentTheme.ts]: Should be const?

File: esp/src/src/UserPreferences/EnvironmentTheme.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        if (val) {
+            context.environmentText.set("value", val);
+            context.environmentTextCB.set("value", true);
+            var parent = domConstruct.create("div", {id: id + "BannerInnerText", class: "envrionmentText"}, searchUserMoreComponents, "before");
```
Comment (GordonSmith) [inline esp/src/src/UserPreferences/EnvironmentTheme.ts]: Should be const?

File: esp/src/src/UserPreferences/EnvironmentTheme.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    }
+}
+
+export function _onResetDefaultTheme(id, context) {
+    ws_store.set("HPCCPlatformWidget_Toolbar_Color", "#2196F3");
+    ws_store.set("HPCCPlatformWidget_Toolbar_Text", "");
+
+    context.toolbarColor.set("value", "#2196F3")
+    context.environmentTextCB.set("checked", false);
+    context.environmentText.set("value", "");
```
Comment (GordonSmith) [inline esp/src/src/UserPreferences/EnvironmentTheme.ts]: context is not valid here

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src/UserPreferences/EnvironmentTheme.ts (hunk 3 above)
Comment (miguelvazq) [reply esp/src/src/UserPreferences/EnvironmentTheme.ts]: Yes I pass it in line 13

File: esp/src/src/UserPreferences/EnvironmentTheme.ts (hunk 6 above)
Comment (miguelvazq) [reply esp/src/src/UserPreferences/EnvironmentTheme.ts]: It is passed in a line `68`

**Comment by GordonSmith** [comment]
@richardkchapman this can be pulled (the let/const comments are very minor).

**Review by GordonSmith — APPROVED** [review]

**Comment by miguelvazq** [comment]
@GordonSmith Modified the commit to reflect the comments that were valid.

**Comment by GordonSmith** [comment]
@richardkchapman good to pull

**Pull request merged**